isobus/include/isobus/isobus/isobus_task_controller_server.hpp

```diff
@@ -0,0 +1,534 @@
+			LocalizationLabel = 0x03, /// /// Sent by the TC or DL to inform the client about the availability of the requested localization version of the device descriptor
+			RequestObjectPoolTransfer = 0x04, /// /// The Request Object-pool Transfer message allows the client to determine whether it is allowed to transfer(part of) the device descriptor object pool to the TC
+			RequestObjectPoolTransferResponse = 0x05, /// Sent in response to Request Object-pool Transfer message
+			ObjectPoolTransfer = 0x06, /// Enables the client to transfer (part of) the device descriptor object pool to the TC
+			ObjectPoolTransferResponse = 0x07, /// Response to an object pool transfer message
+			ObjectPoolActivateDeactivate = 0x08, /// Sent by a client to complete its connection procedure to a TC or DL or to disconnect from a TC or DL.
+			ObjectPoolActivateDeactivateResponse = 0x09, /// Sent by a client to complete its connection procedure to a TC or DL or to disconnect from a TC or DL.
+			DeleteObjectPool = 0x0A, /// This is a message to delete the device descriptor object pool for the client that sends this message.
+			DeleteObjectPoolResponse = 0x0B, /// TC response to a Object-pool Delete message
+			ChangeDesignator = 0x0C, /// This message is used to update the designator of an object.
+			ChangeDesignatorResponse = 0x0D /// Sent in response to Change Designator message
+		};
+
+		/// @brief Stores information about a client that is currently being communicated with.
+		class ActiveClient
+		{
+		public:
+			/// @brief Constructor for an active client object which stores information about a client that is currently being communicated with.
+			/// @param clientControlFunction The control function used to communicate with the client.
+			explicit ActiveClient(std::shared_ptr<ControlFunction> clientControlFunction);
+			std::shared_ptr<ControlFunction> clientControlFunction; ///< The control function used to communicate with the client.
+			std::uint32_t lastStatusMessageTimestamp_ms = 0; ///< The timestamp of the last status message sent to the client.
+			std::uint32_t clientDDOPsize_bytes = 0; ///< The size of the client's DDOP in bytes.
+			std::uint32_t statusBitfield = 0; ///< The status bitfield that the client is reporting to us.
+			std::uint16_t numberOfObjectPoolSegments = 0; ///< The number of object pool segments that have been sent to the client.
+			bool isDDOPActive = false; ///< Whether or not the client's DDOP is active.
+		};
+
+		/// @brief Stores messages received from task controller clients for processing later.
+		/// @param message The message received from the client.
+		/// @param parentPointer A context variable that can be used to find this class's instance.
+		static void store_rx_message(const CANMessage &message, void *parentPointer);
+
+		/// @brief Processes messages received from task controller clients.
+		void process_rx_messages();
```

Comment:
I wasn't really sure what the best way to handle this would be... if we handle the messages on the CAN stack thread, the user needs to know to use aggressive mutexing in their callbacks, but if we do it when `update` is called, then it all happens on their app's thread, which maybe seems nicer? With this approach I only need to mutex internally between `store_rx_message` and `process_rx_messages` (adding this in next commit).

If that seems reasonable to you, I'll add additional comments explaining this

Reply:
Yeah sounds good to me. I think it's a good trade-off to save complexity in the user' application. We can also expose a condition variable to let the application know there are messages available to process, for when one wants to respond asap to received messages without busy-waiting